src/distPlus/distribution/XYShape.re

```diff
@@ -137,6 +143,67 @@ module XtoY = {
       };
     n;
   };
+
+  /* Returns a between-points-interpolating function that can be used with PointwiseCombination.combine.
+     Interpolation can either be stepwise (using the value on the left) or linear. Extrapolation can be `UseZero or `UseOutermostPoints. */
+  let continuousInterpolator = (interpolation: DistTypes.interpolation, extrapolation: DistTypes.extrapolation): interpolator => {
+    switch (interpolation) {
+    | `Linear => switch (extrapolation) {
+      | `UseZero => (t: T.t, leftIndex: int, x: float) => {
+        if (leftIndex < 0) {
```

Comment:
Given that these 4 switch statements follow the same structure, I kind of like the idea of wrapping them in a function that calls a type. Like,
```fun(leftIndex, length) => {
if (leftIndex < 0) { `LeftEdge }
else if (leftIndex >= length - 1) { `RightEdge } 
else { `InBetween }
}
```

Or similar.

Reply:
Right, that does look nice. I am worried about the performance though. This seems to be a relatively consistent bottleneck unfortunately